Assets/Scripts/Zach/InteractorBehaviour.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Zach
+{
+    public class InteractorBehaviour : MonoBehaviour, IInteractor
+    {
+        public IInteractable currentInteraction;
+        public void ReleaseInteraction(IInteractable interactable)
+        {
+            currentInteraction = null;
```

Comment:
you should check to see if the requesting interactable is the currentInteraction before release it